### PR TITLE
Fix for integer overflow for small items like solders

### LINKED ISSUE #26423: Unable to put items with charges into cargo dimension, other items are ok
**Describe the bug**  
Moving items into a cargo dimension works, but moving items with charges like ammo/duct tape gets a message stating the destination is full.

**To Reproduce**  
1. Have a cargo dimension on a vehicle
2. Move an item in
3. Move some ammo in

**Expected behavior**  
Both should be able to work if there is enough space in the cargo dimension

**Screenshots**  
Before moving the gun:
beforemovegun

After moving the gun:
aftermovegun

Trying to move the ammo:
tryingtomoveammo

**Versions and configuration(please complete the following information):**
- First noticed on build 8080, downgraded to 8078 with the same issue. 8076 has no such issue. Suspect it has to do with PR Fix truncated per-charge volume (#26031) in build 8077.

### PR DESCRIPTION
**Pull request by Zetsukaze**
#### Summary
SUMMARY: Bugfixes "Fix for integer overflow for small items like solders"

#### Purpose of change
Fixes #26423 - Unable to put items with charges into cargo dimension
Previous PR was not a comprehensive solution. Further testing caused an integer overflow with putting solders into a cargo dimension.

#### Describe the solution
Changing the return type for item::charges_per_volume to int64_t. This required type casts to other parameters in some functions.

### CODE REVIEW

File: src/activity_handlers.cpp [hunk 1]
```diff
@@ -1342,7 +1342,8 @@ void activity_handlers::fill_liquid_do_turn( player_activity *act_, player *p )
         }
 
         static const auto volume_per_turn = units::from_liter( 4 );
-        const long charges_per_turn = std::max( 1l, liquid.charges_per_volume( volume_per_turn ) );
+        const long charges_per_turn = std::max( static_cast<int64_t>(1l),
```
[the remark this answers is not in the document]
Comment (Zetsukaze) [reply src/activity_handlers.cpp]: Is this the best practice way of doing things?

[the remark this answers is not in the document]
Comment (Qrox) [reply src/activity_handlers.cpp]: I'm not sure how other people think about it, but it does make the code shorter.

**Comment by Zetsukaze** [comment]
Could someone check on this part of the code in catabindings.cpp? charges_per_volume is now int64_t, will the function autocast to int with no integer overflow issue? Or should the function return type be changed as well?
```lua
static int func_item_charges_per_volume(lua_State *L) {
    item& instance = LuaValueOrReference<item>::get(L, 1);
    LuaValue<volume>::check(L, 2);
    auto && parameter1 = LuaValue<volume>::get(L, 2);
    if(lua_gettop(L) > 2) {
        return luaL_error(L, "Too many arguments to charges_per_volume, expected only 2, got %d", lua_gettop(L));
    }
    LuaType<int>::push(L, instance.charges_per_volume(parameter1));
    return 1; // 1 return values
}
```

**Comment by Qrox** [comment]
From `lua/class_definitions.lua`:

> Types can be built in types: int, float, bool, string (std::string in C++), cstring (const char* in C++)
Or any class defined in the classes table.

So I guess the return type cannot be changed to int64_t without modification to the Lua code.

**Comment by Zetsukaze** [comment]
Is it ok to just cast the item charges portions of the code? Changed the liquid ones back to long.

**Comment by jbytheway** [comment]
I think the smallest possible change that would fix putting small objects into large volumes would be to leave the `charges_per_volume` return type unchanged but saturate any value too large for a `long`, replacing it with `INFINITE_CHARGES` (which is the max `long` value).

(Personally I'm of the opinion that all the `long` types in the codebase should be removed in favour of either `int` or `int64_t`, because they just lead to platform-specific bugs that are hard to understand, but that's probably a discussion for another time).

**Pull request closed without merging**